File: Problems/001_TwoSum/main.zig
const std = @import("std");

fn twoSum(num: [4]u8, target: u8) [2]u8 {
    var indNum: [2]u8 = [2]u8{ 0, 0 };

    var i: u8 = 0;
    while (i < num.len) : (i += 1) {
        var j: u8 = i;

        while (j < num.len) : (j += 1) {
            if (i == j) continue;

            if (num[i] + num[j] == target) {
                indNum[0] = i;
                indNum[1] = j;
            }
        }
    }

    return indNum;
}

pub fn main() void {
    const numbers = [_]u8{ 2, 7, 11, 15 };
    const target = 9;
    const result = twoSum(numbers, target);

    std.debug.print("{any}", .{result});
}

File: Problems/001_TwoSum/main.cpp

#include <iostream>
#include <vector>

class Solution {
public:
    std::vector<int> twoSum(const std::vector<int>& nums, int target) {
        std::vector<int> result; 

        for (size_t i = 0; i < nums.size(); i++) {
            for (size_t j = i + 1; j < nums.size(); j++) { 
                if (nums[i] + nums[j] == target) {
                    result.push_back(i); 
                    result.push_back(j); 
                }
            }
        }

        return result;
    }
};

int main() {
    Solution solution; 
    std::vector<int> numbers = {2, 7, 11, 15}; 
    int target = 9; 

    std::vector<int> resultado = solution.twoSum(numbers, target); 
    std::cout << "Índices encontrados: " << resultado[0] << ", " << resultado[1] << std::endl;
    
    return 0;
}

